src/Functions/FunctionSQLJSON.h

```diff
@@ -0,0 +1,332 @@
+                    ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+            }
+            /// Check 1 argument: must be const (JSONPath)
+            if (!isColumnConst(*first_column.column))
+            {
+                throw Exception("1 argument (JSONPath) must be const", ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+            }
+
+            const auto & second_column = arguments[1];
+
+            /// Check 2 argument: must be of type String (JSON)
+            if (!isString(second_column.type))
+            {
+                throw Exception(
+                    "JSONPath functions require 2 argument to be JSON of string, illegal type: " + second_column.type->getName(),
+                    ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+            }
+
+            const ColumnPtr & arg_jsonpath = first_column.column;
+            const auto * arg_jsonpath_const = typeid_cast<const ColumnConst *>(arg_jsonpath.get());
+            const auto * arg_jsonpath_string = typeid_cast<const ColumnString *>(arg_jsonpath_const->getDataColumnPtr().get());
+
+            const ColumnPtr & arg_json = second_column.column;
+            const auto * col_json_const = typeid_cast<const ColumnConst *>(arg_json.get());
+            const auto * col_json_string
+                = typeid_cast<const ColumnString *>(col_json_const ? col_json_const->getDataColumnPtr().get() : arg_json.get());
+
+            /// Get data and offsets for 1 argument (JSONPath)
+            const ColumnString::Chars & chars_path = arg_jsonpath_string->getChars();
+            const ColumnString::Offsets & offsets_path = arg_jsonpath_string->getOffsets();
+
+            /// Prepare to parse 1 argument (JSONPath)
+            const char * query_begin = reinterpret_cast<const char *>(&chars_path[0]);
+            const char * query_end = query_begin + offsets_path[0] - 1;
+
+            /// Tokenize query
+            Tokens tokens(query_begin, query_end);
+            /// Max depth 0 indicates that depth is not limited
+            IParser::Pos token_iterator(tokens, parse_depth);
+
+            /// Parse query and create AST tree
+            Expected expected;
+            ASTPtr res;
+            ParserJSONPath parser;
+            const bool parse_res = parser.parse(token_iterator, res, expected);
+            if (!parse_res)
+            {
+                throw Exception{"Unable to parse JSONPath", ErrorCodes::BAD_ARGUMENTS};
+            }
+
+            /// Get data and offsets for 2 argument (JSON)
+            const ColumnString::Chars & chars_json = col_json_string->getChars();
+            const ColumnString::Offsets & offsets_json = col_json_string->getOffsets();
+
+            JSONParser json_parser;
+            using Element = typename JSONParser::Element;
+            Element document;
+            bool document_ok = false;
+
+            /// Parse JSON for every row
```

Comment:
BTW, there are no tests for multiple rows, but only for single string case. Can you, please, add?